target/linux/ramips/image/mt7621.mk

```diff
@@ -261,6 +261,20 @@ define Device/edimax_ra21s
 endef
 TARGET_DEVICES += edimax_ra21s
 
+define Device/edimax_re23s
+  $(Device/uimage-lzma-loader)
+  IMAGE_SIZE := 15680k
+  DEVICE_VENDOR := Edimax
+  DEVICE_MODEL := RE23S
+  DEVICE_ALT0_VENDOR := Edimax
+  DEVICE_ALT0_MODEL := Gemini RE23S
+  IMAGE/sysupgrade.bin := append-kernel | append-rootfs | \
+	edimax-header -s CSYS -m RN76 -f 0x70000 -S 0x01100000 | pad-rootfs | \
+	append-metadata | check-size
```

Comment:
This looks like a combined sysupgrade/factory image. I think the following should be better:

```
  IMAGES += factory.bin
  IMAGE/factory.bin := append-kernel | append-rootfs | \
	edimax-header -s CSYS -m RN76 -f 0x70000 -S 0x01100000 | pad-rootfs | \
	check-size
```

Despite, please check whether we still need to use pad-rootfs in this case.

Reply:
Maybe you can also use the following?

```
  IMAGE/factory.bin := $$(sysupgrade_bin) | check-size | \
	edimax-header -s CSYS -m RN76 -f 0x70000 -S 0x01100000
```

Reply:
None of them works probably because of the `compatible = "edimax,uimage";` in the firmware partition. 
In fact after the boot it can't find the root partition. Probably it will work if I switch to `compatible = "denx,uimage";`. 
Let me know what I should do. :wink:

Reply:
If mine won't work, I don't see why yours would work. I just don't want to have sysupgrade and factory mixed here.

Reply:
I don't see why the factory image would be dependent on append-metadata. The more interesting question would be whether the default sysupgrade works with edimax,uimage ...

Reply:
Sorry, I don't think I explained myself very well.
The OEM interface, during the firmware upgrade, checks for "CSYS" header at the beginning of the image and then writes the entire image (including the header) on the router. So also the sysupgrade image needs this header or it wont work after a sysupgrade because of the `compatible = "edimax,uimage";`.
On second thought, switching to `compatible = "denx,uimage";` won't solve the problem because the factory image must have the CSYS header or it won't be accepted by the OEM interface.
So, in the end, If we want to use the factory image we can use this:
```
  IMAGES += factory.bin
  IMAGE/factory.bin := append-kernel | append-rootfs | \
	edimax-header -s CSYS -m RN76 -f 0x70000 -S 0x01100000 | check-size
```
but we must also use the sysupgrade entry

Reply:
Okay, but you still don't need append-metadata for the factory images, so this could still be split.

Reply:
Yes, sure. I thought that was a waste of space to create two almost identical images just for the "append-metadata". However I just splitted and tested the images. Everything was working except one thing: without the `pad-rootfs` the sysupgrade image would maintain the configuration even if I deselect the checkbox in the firmware upgrade page. So I restored the `pad-rootfs` and tested again and now everything works correctly. :wink: